R/extend_scenario_trajectory.R

```diff
@@ -372,3 +379,91 @@ handle_phase_out_and_negative_targets <- function(data) {
       )
     )
 }
+
+#' Calculate the ratio of the required change in technology that each company
+#' has achieved per technology at the end of the production forecast period.
+#' This ratio will later serve to adjust the net profit margin for companies
+#' that have not built out enough production capacity in increasing technologies
+#' and hence need to scale up production to compensate for their lag in buildout.
+#'
+#' @param data A data frame containing the production forecasts of companies
+#'   (in the portfolio). Pre-processed to fit analysis parameters and after
+#'   conversion of power capacity to generation.
+#' @param start_analysis Numeric. A vector of length 1 indicating the start
+#'   year of the analysis.
+#' @param time_frame Numeric. A vector of length 1 indicating the number of
+#'   years for which forward looking production data is considered.
+#' @param baseline_scenario Character. A vector of length 1 indicating baseline
+#'   scenario
+#'
+#' @noRd
+calculate_proximity_to_target <- function(data,
+                                          start_analysis,
+                                          time_frame,
+                                          baseline_scenario) {
+  production_changes <- data %>%
+    dplyr::filter(
+      dplyr::between(
+        .data$year, .env$start_analysis, .env$start_analysis + .env$time_frame
```

Comment:
I think if also filtering for scenario the calculations could still be further simplified and maybe even do without all the grouping. After all we are then at 1 row per group.
I know that this would become a problem if one allowed e.g. > 1 investor, but I largely doubt that that will ever happen, so in order to make this implementation easy one could also just add a quick check that each "group" holds only 1 line and then do without the groupings..

Reply:
simplified, warnings are gone now